sql/plan/ddl.go

```diff
@@ -118,14 +118,15 @@ type CreateTable struct {
 	chDefs      []*sql.CheckConstraint
 	idxDefs     []*IndexDefinition
 	like        sql.Node
+	temporary   bool
 }
 
 var _ sql.Databaser = (*CreateTable)(nil)
 var _ sql.Node = (*CreateTable)(nil)
 var _ sql.Expressioner = (*CreateTable)(nil)
 
 // NewCreateTable creates a new CreateTable node
-func NewCreateTable(db sql.Database, name string, ifNotExists bool, tableSpec *TableSpec) *CreateTable {
+func NewCreateTable(db sql.Database, name string, ifNotExists bool, temporary bool, tableSpec *TableSpec) *CreateTable {
```

Comment:
Clean this up: use bool alias types instead of bools for these two params, and introduce consts for them

type IfNotExists bool
type TempTable bool

IfNotExists = true
IfNotExistsAbsent = false
IsTempTable = true
IsNotTempTable = false

Reply:
I experimented with this and feels little messier tbh. I'll push to review though and we can reassess

Reply:
nvm